src/app/pages/TokenDashboardPage/TokenTypeCard.tsx

```diff
@@ -0,0 +1,52 @@
+import { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import { SnapshotCard } from '../../components/Snapshots/SnapshotCard'
+import { COLORS } from '../../../styles/theme/colors'
+import { useRequiredScopeParam } from '../../hooks/useScopeParam'
+import { useTokenInfo } from './hook'
+import { useLoaderData } from 'react-router-dom'
+import { EvmTokenType } from '../../../oasis-indexer/api'
+import { TFunction } from 'i18next'
+import { exhaustedTypeWarning } from '../../../types/errors'
+
+export const getTokenTypeName = (t: TFunction, type: EvmTokenType): string => {
+  switch (type) {
+    case 'ERC20':
+      return t('account.ERC20')
+    default:
+      void exhaustedTypeWarning('Unknown token type', type)
+      return '???'
```

Comment:
Empty return would be better I guess. Just adds unnecessary code.

Reply:
I think I will just return the raw type info.